driver-sync/src/main/com/mongodb/client/internal/CryptConnection.java

```diff
@@ -144,25 +146,11 @@ private FieldNameValidator getFieldNameValidator(final SplittablePayload payload
         return new MappedFieldNameValidator(commandFieldNameValidator, rootMap);
     }
 
-    private BsonWriterSettings createBsonWriterSettings() {
-        return new BsonWriterSettings();
-    }
-
-    // TODO
-    // Currently these settings will allow a split point that is potentially too large after encryption occurs. Need to find a way
-    // to allow at least one document to be included in the split even if it is equal in size to maxDocumentSize, but still split at a
-    // much smaller document size for the command document itself.
-    private BsonBinaryWriterSettings createBsonBinaryWriterSettings() {
-        ConnectionDescription connectionDescription = wrapped.getDescription();
-        return new BsonBinaryWriterSettings(connectionDescription.getMaxDocumentSize());
-    }
-
     private MessageSettings createSplittablePayloadMessageSettings() {
-        ConnectionDescription connectionDescription = wrapped.getDescription();
         return MessageSettings.builder()
-                .maxBatchCount(connectionDescription.getMaxBatchCount())
-                .maxMessageSize(connectionDescription.getMaxDocumentSize())
-                .maxDocumentSize(connectionDescription.getMaxDocumentSize())
+                .maxBatchCount(wrapped.getDescription().getMaxBatchCount())
+                .maxMessageSize(MAX_MESSAGE_SIZE)
+                .maxDocumentSize(MAX_DOCUMENT_SIZE)
```

Comment:
IIUC since we were previously using the values from the wrapped connection description, we were using whatever mongod (not mongocryptd) was advertising in isMaster?

Reply:
Yes, mongod.  This code has no access to the mongocryptd client